Amazon.QLDB.Driver.IntegrationTests/utils/IntegrationTestBase.cs

```diff
@@ -106,6 +107,7 @@ public void RunCreateLedger()
 
             return builder.WithQLDBSessionConfig(amazonQldbSessionConfig)
                 .WithLedger(finalLedgerName)
+                .WithSerializer(new ObjectSerializer())
```

Comment:
Is this a requirement, or will it default to this serializer?

Reply:
This is a requirement if the user is planning on using `IQuery`. If the user doesn't pass a serializer, there is a [null check ](https://github.com/awslabs/amazon-qldb-driver-dotnet/blob/object-serialization-idda-jun-4-2021/Amazon.QLDB.Driver/query/Query.cs#L52-L55)in `Query` constructor.

There is an integration [test](https://github.com/awslabs/amazon-qldb-driver-dotnet/blob/object-serialization-idda-jun-4-2021/Amazon.QLDB.Driver.IntegrationTests/StatementExecutionTests.cs#L292-L312) that tests using serializer so it is required here.

Reply:
Right - by why can't we default to `ObjectSerializer` if one isn't set rather than forcing someone to set one?

Reply:
Theres two issues:

* There is a cyclic dependency error
  * `Amazon.QLDB.Driver.Serialization` is using `Amazon.QLDB.Driver`.
  * If `Amazon.QLDB.Driver` uses `Amazon.QLDB.Driver.Serialization` then it cause the error.

* `Amazon.QLDB.Driver` is using target framework: `netstandard 2.0` and it is incompatible with `Amazon.QLDB.Driver.Serialization`/`Amazon.IonObjectMapper` target framework: `net50`

Reply:
Ah - so we have the interface but not the implementation. Noted - too bad this can't be implicitly loaded if the serialization library is in the runtime (or can it?).